chromium_src/chrome/browser/ui/views/tabs/tab_style_views.cc

```diff
@@ -31,8 +33,21 @@ class BraveGM2TabStyle : public GM2TabStyle {
   TabStyle::TabColors CalculateColors() const override;
   const gfx::FontList& GetFontList() const override;
   SeparatorBounds GetSeparatorBounds(float scale) const override;
+  void PaintTab(gfx::Canvas* canvas) const override;
 
  private:
+  bool IsVerticalTab() const {
```

Comment:
I think it would be better to subclass here and return the correct subclass in ```
std::unique_ptr<TabStyleViews> TabStyleViews::CreateForTab(Tab* tab) {
  return std::make_unique<BraveGM2TabStyle>(tab);
}
``` based on whether the feature is enabled or not